src/modules/joke.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * @file
+ * Modules:
+ *  - {@link joke}
+ * Description:
+ *  Uses the joke api provided by arc to parse a response and send it as a discord embed
+ * @throws will throw an error if the api call fails
+ */
+
+import * as util from '../core/util.js';
+import {botConfig} from '../core/config.js';
+
+// Config manager for nsfw detection
+const jokeConfig = botConfig.modules.joke;
+let jokeApiUrl = 'https://v2.jokeapi.dev/joke/Any';
+
+const blacklistFlags = [
+  'religious',
+  'political',
+  'racist',
+  'sexist',
+  'explicit',
+];
+
+if (!jokeConfig.nsfw) {
+  blacklistFlags.push('nsfw');
+}
+
+jokeApiUrl += `?blacklistFlags=${blacklistFlags.join(',')}&type=single`;
+
+const joke = new util.RootModule(
+  'joke',
+  'Get a funny joke from the bot',
+  [],
+  [],
+
+  async (args, interaction) => {
+    const fetchedJoke = await fetchJoke();
+    util.replyToInteraction(interaction, {
+      embeds: [util.embed.infoEmbed(`${fetchedJoke}`)],
+    });
+  }
+);
+
+async function fetchJoke(): Promise<string> {
+  try {
```

Comment:
This try/catch is redundant.

You can also catch the error on line 38 with 
```typescript
await fetchJoke().catch((error) => {
  // run your error handling code here
});
```

Reply:
https://github.com/r-Techsupport/TuringBot/pull/53/commits/d049ce68a93b74d6a25d356502de747bb5797bfe